nixos/modules/hardware/openrazer.nix

```diff
@@ -49,7 +49,11 @@ in
 {
   options = {
     hardware.openrazer = {
-      enable = mkEnableOption "OpenRazer drivers and userspace daemon";
+      enable = mkEnableOption ''
+        OpenRazer drivers and userspace daemon.
+        Any users that wish to make use of the daemon need to be
+        members of the "plugdev" group.
+      '';
```

Comment:
Why don't you add a users option instead? We could manage the group membership based on that option.

Reply:
ah, that's a very good suggestion, I'll take a stab at it, sure

Reply:
i cant find other modules that do this so i dont think it should be done like that

Reply:
I have no strong opinion on the matter.

Reply:
Such an option may have unforeseen security consequences, because `plugdev` may grant access to more than this specific hardware.

Reply:
Indeed, so udev files shipped should use a dedicated group, not plugdev.

I reiterate that sentiment, but plugdev is a debianism, that doesn't even match for most use cases, which is why I prefer we create more specific groups instead.

> plugdev: Allows members to mount (only with the options nodev and nosuid, for security reasons) and umount removable devices through pmount. 

https://wiki.debian.org/SystemGroups

Reply:
That is also a good point. My general feelings on this are:
- I could ask upstream to change the group name, but that is probably _too_ breaking a change for them
- Patching it for nix(os) is not particularly friendly towards upstream, since any nixos setups will have different groups, and I'd expect that not to be an obvious difference.
- it's not particularly great security wise to give users umbrella permissions to a group just to tweak RGB lights

I wonder how easy it would be for upstream to support both plugdev and a new. unique group. I can open an issue there for that. That would be the ideal solution imo.

Reply:
It would be easy enough to substitute the distinctive term `plugdev` with `openrazer` in the package build without involving the upstream. The chances for upstream to consider this are slim to none IMO.

Reply:
indeed. I will wait for everyone else voice their opinion on this, and will implement what everyone else things is best

Reply:
Changed the docs to mention the openrazer group, and fixed up the various openrazer packages and module to use the new group. I also changed the branch this PR is for to staging, the docs said that should be done for breaking changes. Let me know if I misunderstood

Reply:
Just saw this change now and was initially confused about the intent/cause for this change. 

Do we already specify _why_ this change is required in the changelog? 

And if not, might including a note regarding said decision help give some context for the change that wouldn’t require referring to this issue?

e.g. “the openrazer … modules now require the `openrazer` group rather than `plugdev` because/so that…”

Side note: I have no experience with security wrappers, but I do wonder if we could handle this like the [wireshark module](https://github.com/NixOS/nixpkgs/blob/6b940846e99cedd10eafbc301e467972ade1afec/nixos/modules/programs/wireshark.nix#L34) does.

Reply:
I have added the reasoning behind the change to the documentation. 

As for security wrappers, I don't know either, never had to look at anything like that. From a cursory read around docs on the internet, maybe there is something that could be done with them in the openrazer-daemon side, but I don't think there is anything to do in linuxPackages.openrazer. I'd have to find some time to properly research this to see what might be doable or even if it makes sense to use security.wrappers.

Reply:
Thank you! It’s probably fine as is for now, I was more just curious about it in general. 

This specific problem doesn’t seem too specific to openrazer is all, and I wonder how many packages might benefit from a solution similar to what this PR offers.

Reply:
Finally, we could create the `hardware.openrazor.users` option.

Reply:
Does this abstraction work for networked users provided by `ldap`, etc...?

Reply:
Also, please removing trailing period because `mkEnableOption` adds a trailing period.

Reply:
> Does this abstraction work for networked users provided by `ldap`, etc...?

If they propagate a PosixGroup of that name it would.